### ISSUE TITLE
Installing beaker browser on Xubuntu 32 bit

### ISSUE THREAD

**Issue by pandafy** [beakerbrowser/beaker#1080]
Operation System:Xubuntu
I am wondering if it is possible to install beaker browser on 32bit OS as AppImage for Linux says x86_64.
@pfrazee 
```
$ ./beaker-browser-0.8.0-prerelease.6-x86_64.AppImage 
bash: ./beaker-browser-0.8.0-prerelease.6-x86_64.AppImage: cannot execute binary file: Exec format error

```

**Comment by pfrazee** [comment]
@TheOneAboveAllTitan I think you'll probably have to build from source for now, I'm afraid

**Comment by TheOneAboveAllTitan** [comment]
I got the following output after running npm start. Is it usual?
```
~/Beaker_Browser/beaker$ npm start

> beakerbrowser@ start /home/stark/Beaker_Browser/beaker
> gulp start

[13:21:11] Using gulpfile ~/Beaker_Browser/beaker/gulpfile.js
[13:21:11] Starting 'start'...
Spawning electron /home/stark/Beaker_Browser/beaker/node_modules/electron/dist/electron
[13:21:11] Finished 'start' after 25 ms
dat-serve Failed to load sodium-native. Performance may be degraded. Error: Cannot find module 'sodium-native'

Failed to load sodium-native. Performance may be degraded. { Error: Cannot find module 'sodium-native'
    at Module._resolveFilename (module.js:543:15)
    at Function.Module._resolveFilename (/home/stark/Beaker_Browser/beaker/node_modules/electron/dist/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:473:25)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/stark/Beaker_Browser/beaker/app/node_modules/@beaker/core/dat/protocol.js:27:3)
    at Object.<anonymous> (/home/stark/Beaker_Browser/beaker/app/node_modules/@beaker/core/dat/protocol.js:352:3)
    at Module._compile (module.js:642:30)
    at Object.Module._extensions..js (module.js:653:10)
    at Module.load (module.js:561:32) code: 'MODULE_NOT_FOUND' }
Logfile: /home/stark/.config/Beaker Browser/debug.log
Logfile: /home/stark/.config/Beaker Browser/debug.log

```

**Comment by pfrazee** [comment]
It's not ideal but not a problem either (will only affect performance).

Try running `npm run rebuild`

**Comment by TheOneAboveAllTitan** [comment]
Still getting the same error output

**Comment by brechtcs** [comment]
After a couple of failed attempts a few months ago, I managed to build an AppImage for 32bit Linux today. If you want I could make one for the latest release (and any upcoming releases) so it can be added to the Releases page.

**Comment by pfrazee** [comment]
@brechtcs glad to hear, what did you end up changing to support it?

**Comment by brechtcs** [comment]
Well, for starters I'm working on an i386 system, so that probably makes
things significantly easier for me.

Apart from that there's two things I had to pay special attention to:

- The node version seems to matter a lot. With the wrong version even `npm
start` doesn't work. Before the electron update I was using node 8, node 9
didn't work. For the current master 10.x didn't work for me, but the latest
11.x does.
- Instead of `npm run release` I'm calling electron-build directly like
this: `node_modules/.bin/build -l AppImage`.

On a 64bit system you might have to add an extra architecture flag for i386
to the electron-build command. But I'm not sure that alone will suffice,
because earlier in the build process all native dependecies (sodium,
leveldb, sqlite, the lot) are compiled specifically for my system.

Basically, given the difficulties I had on my 32bit system, I fear the
worst for any cross-compilation workflow from a 64bit machine. I hope I'm
wrong.

If I am right though, what do you think would be the best way to proceed? I
would understand if you're reluctant to simply drop my builds into the
official release channel. Maybe it would be better if I set up and maintain
a separate semi-official release channel?

Op ma 12 nov. 2018 16:06 schreef Paul Frazee <notifications@github.com:

**Comment by pfrazee** [comment]
For now, I think just having a build from source that works for @TheOneAboveAllTitan would be a good step. @TheOneAboveAllTitan Can you try what @brechtcs suggests and lmk how it goes?
